### PR TITLE
Support LSM6DSV IMU

### PR DESCRIPTION
**Pull request by wigwagwent**
This adds support for the lsm6dsv family of IMUs. They are better than bmi160 and seems comparable to the bno but need more users testing them.

This merge should probably be squashed into 1 commit.

### CODE REVIEW

File: src/consts.h [hunk 1]
```diff
@@ -35,6 +35,7 @@
 #define IMU_BMI160 8
 #define IMU_ICM20948 9
 #define IMU_ICM42688 10
+#define IMU_LSM6DSV 12
```
Comment (ButterscotchV) [inline src/consts.h:38]: This should be noted when merging, depending on the order of merging PRs this might be awkward to skip a number.

**Review by ButterscotchV — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -34,8 +34,15 @@ The following IMUs and their corresponding `IMU` values are supported by the fir
 * ICM-20948 (IMU_ICM20948)
   * Using fusion in internal DMP for 6Dof or 9DoF, 9DoF mode requires good magnetic environment.
   * Comment out `USE_6DOF` in `debug.h` for 9DoF mode.
+  * Experimental support!\
```
Comment (TheDevMinerTV) [inline README.md:37]: Something broke here, what's that `\`?

File: lib/LSM6DSV/LSM6DSV.cpp [hunk 3]
```diff
@@ -0,0 +1,4080 @@
+/**
```
Comment (TheDevMinerTV) [inline lib/LSM6DSV/LSM6DSV.cpp:1]: Please include a link to the original repository.

File: src/sensors/SensorManager.cpp [hunk 4]
```diff
@@ -108,6 +109,12 @@ namespace SlimeVR
             case IMU_ICM42688:
                 sensor = new ICM42688Sensor(sensorID, address, rotation, sclPin, sdaPin);
                 break;
+            case IMU_LSM6DSV:
+                {
+                uint8_t intPin = extraParam;
+                sensor = new LSM6DSVSensor(sensorID, imuType, address, rotation, sclPin, sdaPin, intPin);
+                }
+                break;
```
Comment (TheDevMinerTV) [inline src/sensors/SensorManager.cpp:117]:
```suggestion
                sensor = new LSM6DSVSensor(sensorID, imuType, address, rotation, sclPin, sdaPin, extraParam);
                break;
```

File: lib/math/quat.h [hunk 5]
```diff
@@ -72,7 +72,7 @@ class Quat {
 	Vector3 get_euler_yxz() const;
 
 	void set_euler(const Vector3& p_euler) { set_euler_yxz(p_euler); };
-	Vector3 get_euler() const { return get_euler_yxz(); };
+	Vector3 get_euler() const { return get_euler_xyz(); };
```
Comment (TheDevMinerTV) [inline lib/math/quat.h:75]: Instead of modifying `get_euler()` you should rather use the `get_euler_xyz()` in your code.

**Review by TheDevMinerTV — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -34,8 +34,15 @@ The following IMUs and their corresponding `IMU` values are supported by the fir
 * ICM-20948 (IMU_ICM20948)
```
Comment (TheDevMinerTV) [inline README.md:34]: GitHub doesn't allow me to request changes without a comment...

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

**Review by Eirenliel — CHANGES_REQUESTED** [review]
Please address @TheDevMinerTV's comments

File: src/sensors/SensorManager.cpp (hunk 4 above)
Comment (Pespiri) [reply src/sensors/SensorManager.cpp:117]: I would suggest against this change, as the original change follow the pattern of how other cases in this switch statement has done this

image

**Comment by unlogisch04** [comment]
Is this still needed if we go for #322 ?

**Comment by gorbit99** [comment]
> Is this still needed if we go for #322 ?

This probably shouldn't be preferred over sfusion, at least not without a cleanup.

**Comment by unlogisch04** [comment]
I convert that to draft, as sfusion seems to be the better solution.
@wigwagwent feel free to close it when you see it the same way.

**Pull request closed without merging**